### DISCUSSION TITLE
Map List and flatten it

### DISCUSSION THREAD

**Discussion by fpetrov** [riok/mapperly#1195] [Q&A] [answered]
I have these classes:

`public class Trip
{
    // ...
    public List<User> Subscribers { get; set; } = [];
    // ...
}`

`
public class User
{
    public long Id { get; set; }
}
`

`
public record TripResult(
    // ...
    List<long> SubscribersIds
    // ...);
`

So, I want to map all **Trip Subscribers** to **TripResult SubscribersIds** like: trip.Subscribers.**ConvertAll**(s => s.Id)

Is it possible to implement in Mapperly?
Thanks

**Comment by TimothyMakkison** [comment, marked as answer]
I'd recommend using a [before/after map](https://mapperly.riok.app/docs/configuration/before-after-map/) for this.

```c#
[Mapper]
public static class MyMapper
{
    public static TripResult MapTrip(Trip source)
    {
        var result = InternalMapTrip(source);
        result.SubscribersIds = source.Subscribers.ConvertAll(s => s.Id);
       return result;
    }
    
    [MapperIgnoreTarget("SubscribersIds")]
    public static partial TripResult InternalMapTrip(Trip source);
}
```
